### PR TITLE
Added unit tests for property fields

### PR DESCRIPTION
**Pull request by renatgalimov**
#27 has a huge refactoring, so I want to have unit tests in place for the old code to ensure that I don't break anything with the refactoring. 

In addition - this PR fixes unit-tests for emacs 26

### CODE REVIEW

**Comment by renatgalimov** [comment]
I will try to fix test for emacs 26

File: anki-editor-tests.el [hunk 1]
```diff
@@ -168,5 +169,49 @@ Simple note body
...
+      (unwind-protect
+          (should (equal
+                   (anki-editor-note-at-point)
+                   #s(anki-editor-note nil "Basic" "Tests"
+                                       (("Back" . "<p>\nShould be included\n</p>\n")
+                                        ("Front" . "<p>\nCan one define an anki-field inside an org-mode property?</p>\n"))
+                                       nil)))
+        (anki-editor-test--teardown)))))
+
+(ert-deftest test--node-at-point-for-note-with-property-field-should-override-subheading-field ()
```
Comment (slotThe) [inline anki-editor-tests.el:187]: Mh, this fails for me and instead includes the test from the `Front` subheading

File: anki-editor.el [hunk 2]
```diff
@@ -819,9 +819,13 @@ and else from variable `anki-editor-prepend-heading'."
...
-    (if anki-editor-org-tags-as-anki-tags
-        (append tags (mapcar #'substring-no-properties (org-get-tags)))
-      tags)))
+    (when anki-editor-org-tags-as-anki-tags
+      (setq tags (append tags (mapcar #'substring-no-properties (org-get-tags)))))
+
+    ;; Filtering out empty tags produces by org-mode coming with emacs 26
+    (cl-loop for tag in tags
+             if (and tag (not (string= tag "")))
+             collect tag)))
```
[the remark this answers is not in the document]
Comment (renatgalimov) [reply anki-editor.el:828]: I agree. I might remove emacs 26 tests from Actions, but this block makes sense as we filter out empty fields.

[the remark this answers is not in the document]
Comment (slotThe) [reply anki-editor.el:828]: I agree that filtering out empty fields makes sense :+1: Seeing the "Emacs 26" comment just reminded me of this. Let's just officially drop that then

File: anki-editor-tests.el [hunk 3]
```diff
@@ -168,5 +169,49 @@ Simple note body
         (anki-editor-test--teardown)))))
 
 
+(ert-deftest test--node-at-point-for-note-with-property-field-should-render-property-field ()
```
Comment (slotThe) [inline anki-editor-tests.el:172]:
```suggestion
(ert-deftest test--note-at-point-for-note-with-property-field-should-render-property-field ()
```

File: anki-editor-tests.el (hunk 1 above)
Comment (slotThe) [inline anki-editor-tests.el:187]:
```suggestion
(ert-deftest test--note-at-point-for-note-with-property-field-should-override-subheading-field ()
```

File: anki-editor-tests.el [hunk 4]
```diff
@@ -168,5 +169,49 @@ Simple note body
...
+          (should (equal
+                   (anki-editor-note-at-point)
+                   #s(anki-editor-note nil "Basic" "Tests"
+                                       (("Back" . "<p>\nShould be included\n</p>\n")
+                                        ("Front" . "<p>\nCan one define an anki-field inside an org-mode property?</p>\n"))
+                                       nil)))
+        (anki-editor-test--teardown)))))
+
+
+(ert-deftest test--node-at-point-for-note-with-unknown-property-field-should-raise-error ()
```
Comment (slotThe) [inline anki-editor-tests.el:203]:
```suggestion
(ert-deftest test--note-at-point-for-note-with-unknown-property-field-should-raise-error ()
```

File: anki-editor.el (hunk 2 above)
[the remark this answers is not in the document]
Comment (renatgalimov) [reply anki-editor.el:828]: @slotThe , in another PR I moved our CI from emacs 26 and 27 to emacs 28.1 and emacs 29.3

[the remark this answers is not in the document]
Comment (slotThe) [reply anki-editor.el:828]: Okay, I think this is good to go then!

**Comment by slotThe** [comment]
Thanks!

**Pull request merged**